WPFHospitalEditor/SpecialistAppointmentSearchDialog.xaml.cs

```diff
@@ -0,0 +1,135 @@
+            }
+            
+        }
+
+        private void loadEquipment()
+        {
+            List<EquipmentTypeDto> equipmentTypes = equipmentTypeServerController.GetAllEquipmentTypes().ToList();
+            
+            foreach(EquipmentTypeDto e in equipmentTypes)
+            {
+                equipmentComboBox.Items.Add(e.Name);
+            }
+        }
+
+        private void Close_Click(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void Search_Click(object sender, RoutedEventArgs e)
+        {
+            HospitalMap.clearAllResults();
+            if (InvalidInputForAppointment())
+            {
+                MessageBox.Show("Invalid input.");
+            }
+            else
+            {
+                String specialist = specialistComboBox.SelectedItem.ToString();
+                int specialistID = int.Parse(specialist.Split(" ")[0]);
+                Doctor chosenDoctor = doctorServerController.GetDoctorById(specialistID);
+                DateTime startDate = DateTime.ParseExact(dateFrom.Text + " 08:00", "MM/dd/yyyy HH:mm", null);
+                DateTime endDate = DateTime.ParseExact(dateTo.Text + " 16:00", "MM/dd/yyyy HH:mm", null);
+
+                TimeInterval timeInterval = new TimeInterval(startDate, endDate);
+                RecommendationPreference appointmentSearchPriority = setPriorityFromComboBox();
+
+                RecommendationRequestDto recommendationRequestDto = new RecommendationRequestDto()
+                {
+                    DoctorId = chosenDoctor.Id,
+                    SpecialtyId = chosenDoctor.DepartmentId,
+                    TimeInterval = timeInterval,
+                    Preference = appointmentSearchPriority
+                };
+
+                if (CheckEquipmentExistance())
+                {
+                    HospitalMap.appointmentSearchResult = schedulingServerController.GetAppointments(recommendationRequestDto);
+                    SearchResultDialog appointmentDialog = new SearchResultDialog(hospitalMap, SearchType.AppointmentSearch);
+                    appointmentDialog.ShowDialog();
+                }
+                else
+                {
+                    MessageBox.Show("There is no room with required equipment!");
+                }
+
+                
+            }
+        }
+
+        private bool InvalidInputForAppointment()
+        {
+            if (specialistComboBox.Text.Equals("") || dateFrom.Text.Equals("") || dateTo.Text.Equals("") || equipmentComboBox.Text.Equals("") || priorityComboBox.Text.Equals("")) return true;
+            return false;
+        }
+
+        private RecommendationPreference setPriorityFromComboBox()
+        {
+            if (priorityComboBox.SelectedIndex == 0) return RecommendationPreference.Doctor;
+            return RecommendationPreference.Time;
+        }
```

Comment:
Have same method in HospitalMap. That method is using priorityComboBox from hospital map. Try to refactor this.

Reply:
Additionally, you have named function "setPriorityFromComboBox", but this function doesn't change any data.